src/Symfony/Component/HttpKernel/Controller/ArgumentResolver.php

```diff
@@ -45,8 +45,9 @@ public function __construct(ArgumentMetadataFactoryInterface $argumentMetadataFa
     public function getArguments(Request $request, callable $controller): array
     {
         $arguments = [];
+        $reflectors = $request->attributes->get('_controller_reflectors') ?? [];
```

Comment:
if `$event->setAttributes` is used, I don't see where they would be assigned there

Reply:
They would always be assigned, because ControllerEvent::setController is always called, and this is independent from using setAttributes or not. Or do you mean something else?

Reply:
but if `setController` is called before `setAttributes` (likely to be the case if a listener is the one calling `setAttributes` to set custom attributes), the assignment done in `setController` will not include those custom attributes.
To me, the kernel should be the one doing that assignment *after* triggering the event, based on `$event->getAttributes`